File: Data/Sys/LuaLibs/instruction_step.lua
dolphin:importModule("InstructionStepAPI", "1.0")

InstructionStepClass = {}

function InstructionStepClass:new(obj)
	obj = obj or {}
	self.__index = self
	return setmetatable(obj, self)
end

instruction_step = InstructionStepClass:new(nil)

function InstructionStepClass:singleStep()
	InstructionStepAPI:singleStep()
end

function InstructionStepClass:stepOver()
	InstructionStepAPI:stepOver()
end

function InstructionStepClass:stepOut()
	InstructionStepAPI:stepOut()
end

function InstructionStepClass:skip()
	InstructionStepAPI:skip()
end

function InstructionStepClass:setPC(newPCValue)
	InstructionStepAPI:setPC(newPCValue)
end

function InstructionStepClass:getInstructionFromAddress(address)
	return InstructionStepAPI:getInstructionFromAddress(address)
end

File: Data/Sys/PythonLibs/instruction_step.py
import dolphin
dolphin.importModule("InstructionStepAPI", "1.0")
import InstructionStepAPI

def singleStep():
    InstructionStepAPI.singleStep()

def stepOver():
    InstructionStepAPI.stepOver()

def stepOut():
    InstructionStepAPI.stepOut()

def skip():
    InstructionStepAPI.skip()

def setPC(newPCValue):
    InstructionStepAPI.setPC(newPCValue)

def getInstructionFromAddress(address):
    return InstructionStepAPI.getInstructionFromAddress(address)

